src/Verifier/SAW/Simulator/What4.hs

```diff
@@ -557,6 +558,23 @@ arrayUpdate sym arr idx elm
   | otherwise =
     panic "Verifier.SAW.Simulator.What4.Panic.arrayUpdate" ["argument type mismatch"]
 
+arrayEq ::
+  W.IsSymExprBuilder sym =>
+  sym ->
+  SArray sym ->
+  SArray sym ->
+  IO (W.Pred sym)
+arrayEq sym lhs_arr rhs_arr
+  | SArray lhs_arr_expr <- lhs_arr
+  , SArray rhs_arr_expr <- rhs_arr
+  , W.BaseArrayRepr (Ctx.Empty Ctx.:> lhs_idx_repr) lhs_elm_repr <- W.exprType lhs_arr_expr
+  , W.BaseArrayRepr (Ctx.Empty Ctx.:> rhs_idx_repr) rhs_elm_repr <- W.exprType rhs_arr_expr
+  , Just Refl <- testEquality lhs_idx_repr rhs_idx_repr
+  , Just Refl <- testEquality lhs_elm_repr rhs_elm_repr =
+    W.arrayEq sym lhs_arr_expr rhs_arr_expr
+  | otherwise =
+    panic "Verifier.SAW.Simulator.What4.Panic.arrayEq" ["argument type mismatch"]
```

Comment:
Is there any way user input can trigger this panic, or is it only possible if we have a bug?

Reply:
only in case of a bug